synapse/rest/client/v1/room.py

```diff
@@ -222,14 +224,26 @@ def register(self, http_server):
     async def on_POST(self, request, room_id, event_type, txn_id=None):
         requester = await self.auth.get_user_by_req(request, allow_guest=True)
         content = parse_json_object_from_request(request)
+        prev_events = parse_strings_from_args(request.args, "prev_event")
+        origin_server_ts = parse_integer(request, "origin_server_ts")
 
         event_dict = {
             "type": event_type,
             "content": content,
             "room_id": room_id,
-            "sender": requester.user.to_string(),
+            "sender": requester.user.to_string()
         }
 
+        if prev_events:
+            event_dict["prev_events"] = prev_events
+
+        # TODO: Add `and requester.app_service`
+        if origin_server_ts:
+            event_dict["origin_server_ts"] = origin_server_ts
+
+        # TODO: I noticed in the Synapse code that we already accept a `ts` query parameter to override
+        # the `origin_server_ts` if the request is coming from an app service.
+        # Do we want to remove in favor of the spec'ed code above
```

Comment:
MSC2716 doesn't add a `origin_server_ts` query param, it just re-adds the `ts` query param to the spec.

Reply:
Ahh, mis-read `origin_server_ts` vs `ts` [there](https://github.com/matrix-org/matrix-doc/pull/2716/files#r567176414). Thanks @tulir!